iron-doc-viewer.html

```diff
@@ -195,6 +211,34 @@
       this._behaviors  = this.descriptor.behaviors || [];
 
       this.toggleAttribute('abstract', this.descriptor.abstract);
+
+      // On initial load, force position to the correct anchor.
+      // (This shouldn't be required in the catalog, but the catalog
+      // may scroll before content is fully loaded, so we try again here.)
+      this.scrollToAnchor();
+    },
+
+    /**
+     * Scrolls to the currently selected anchor, as identified
+     * by the URL hash. Must called from the outside if the
+     * hash changes but the selected element doesn't (for example,
+     * the user clicks on a property name on the current page).
+     */
+    scrollToAnchor: function() {
+      // scroll to anchor after data finishes loading
+      if (window.location.hash) {
+        // ToDo: this async appears to be required
+        // on catalog, but it's unclear why.
+        this.async(function() {
```

Comment:
Is that change merged? I missed it somehow.

Mine handles setting active, which is important for deep-linking, but we could certainly reconcile the two implementations. I did it in iron-doc-viewer to avoid using a /deep/ query.

The "deep" solution he's using won't work with my version because (for the static routing case) I was using `:` as a divider between the element name (`Polymer.Base`, `dom-repeat`, etc.) and the rest of the ID, and that makes it an invalid selector.

One alternative would be to use a different safe character, but the only thing that occurred to me was a double dash (so, `Polymer.Base--method-debounce`). The separator can't be anything that can occur in an element name, nor anything that has special significance in a CSS selector, like `:` or `.`. 

Using /deep/ in all cases would simpify the implementation, but long-term it's a dead end. Thoughts, @ebidel?